File: frontend/src/types/property.ts
export interface PropertySummary {
  id: string
  url: string
  title?: string
  address?: string
  dublin_district?: string
  price?: number
  bedrooms?: number
  bathrooms?: number
  carpet_area_sqm?: number
  property_type?: string
  ber_rating?: string
  heating_type?: string
  is_chain_free?: boolean
  is_south_facing?: boolean
  is_htb_eligible?: boolean
  days_on_market?: number
  estate_agent?: string
  latitude?: number
  longitude?: number
  is_active: boolean
  missing_data_flags?: string[]
}

export interface NeighbourhoodScore {
  safety_score?: number
  amenities_score?: number
  connectivity_score?: number
  schools_score?: number
  parks_score?: number
  cafes_score?: number
  supermarkets_score?: number
  m50_n11_score?: number
  overall_score?: number
  commute_dundrum_min?: number
}

export interface PropertyDetail extends PropertySummary {
  eircode?: string
  year_built?: number
  management_fee_eur?: number
  seller_status?: string
  description?: string
  features_list?: string[]
  price_history?: { price: number; date: string; type: string }[]
  neighbourhood_score?: NeighbourhoodScore
}

export interface SearchFilters {
  price_min?: number
  price_max?: number
  bedrooms_min?: number
  bathrooms_min?: number
  carpet_area_sqm_min?: number
  property_types?: string[]
  ber_ratings?: string[]
  heating_types?: string[]
  dublin_districts?: string[]
  is_chain_free?: boolean
  is_htb_eligible?: boolean
  days_on_market_max?: number
  exclude_electric_storage?: boolean
}

export interface SearchRequest {
  query: string
  filters?: SearchFilters
  limit?: number
  offset?: number
}

export interface SearchResponse {
  results: PropertySummary[]
  total: number
  search_id: string
  query_time_ms: number
}

export interface CompareResponse {
  summary: string
  properties: PropertySummary[]
  comparison_table: Record<string, Record<string, unknown>>
}

File: backend/app/schemas/property.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class NeighbourhoodScoreOut(BaseModel):
    safety_score: float | None = None
    amenities_score: float | None = None
    connectivity_score: float | None = None
    schools_score: float | None = None
    parks_score: float | None = None
    cafes_score: float | None = None
    supermarkets_score: float | None = None
    m50_n11_score: float | None = None
    overall_score: float | None = None
    commute_dundrum_min: float | None = None
    computed_at: datetime | None = None


class PropertySummary(BaseModel):
    id: uuid.UUID
    url: str
    title: str | None = None
    address: str | None = None
    dublin_district: str | None = None
    price: int | None = None
    bedrooms: int | None = None
    bathrooms: int | None = None
    carpet_area_sqm: int | None = None
    property_type: str | None = None
    ber_rating: str | None = None
    heating_type: str | None = None
    is_chain_free: bool | None = None
    is_south_facing: bool | None = None
    is_htb_eligible: bool | None = None
    days_on_market: int | None = None
    estate_agent: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    is_active: bool = True
    missing_data_flags: list[str] | None = None

    model_config = {"from_attributes": True}


class PropertyDetail(PropertySummary):
    eircode: str | None = None
    year_built: int | None = None
    management_fee_eur: int | None = None
    seller_status: str | None = None
    description: str | None = None
    features_list: list[str] | None = None
    price_history: list[dict] | None = None
    neighbourhood_score: NeighbourhoodScoreOut | None = None


class CompareRequest(BaseModel):
    property_ids: list[uuid.UUID] = Field(..., min_length=2, max_length=3)


class CompareResponse(BaseModel):
    summary: str
    properties: list[PropertySummary]
    comparison_table: dict
